extensions/amp-form/0.1/amp-form.js

```diff
@@ -705,42 +704,47 @@ function isDisabled_(element) {
 
 
 /**
- * Installs submission handler on all forms in the document.
- * @param {!Window} win
+ * Bootstraps the amp-form elements
  */
-function installSubmissionHandlers(win) {
-  onDocumentReady(win.document, doc => {
-    toArray(doc.forms).forEach((form, index) => {
-      if (!form.classList.contains('i-amphtml-form')) {
-        new AmpForm(form, `amp-form-${index}`);
-      }
+export class AmpFormService {
+  /**
+   * @param  {!../../../src/service/ampdoc-impl.AmpDoc} ampdoc
+   */
+  constructor(ampdoc) {
+    const doc = ampdoc.win.document;
+    installStyles(doc, CSS, () => {
```

Comment:
Hmm. Even `installStyles` is an issue :( It has to be:
 - `installStyles` for a single doc
 - `installStyles` for FIE
 - `installStylesForShadowRoot` for a shadow doc